lib/string_reverse.rb

```diff
@@ -2,5 +2,19 @@
 # Time complexity: ?
 # Space complexity: ?

```

Comment:
Don't forget time/space complexity!

Here's how I answered these. This is about as thorough as I can think to be in analyzing space & time complexity: https://github.com/mmcknett/string_reverse/blob/implementation/lib/string_reverse.rb#L3-L11